lib/src/impl_ffi/impl_ffi.digest.dart

```diff
@@ -62,39 +62,104 @@ abstract class _Hash implements Hash {
       return out.copy(size);
     });
   }
+
+  @override
+  String hmacJwkAlg(HashImpl hash) {
+    if (hash == sha1) {
+      return 'HS1';
+    } else if (hash == sha256) {
+      return 'HS256';
+    } else if (hash == sha384) {
+      return 'HS384';
+    } else if (hash == sha512) {
+      return 'HS512';
+    } else {
+      throw UnsupportedError('hash is not supported');
+    }
+  }
+
+  @override
+  String rsaOaepJwkAlg(HashImpl hash) {
+    if (hash == sha1) {
+      return 'RSA-OAEP-1';
+    } else if (hash == sha256) {
+      return 'RSA-OAEP-256';
+    } else if (hash == sha384) {
+      return 'RSA-OAEP-384';
+    } else if (hash == sha512) {
+      return 'RSA-OAEP-512';
+    } else {
+      throw UnsupportedError('hash is not supported');
+    }
+  }
+
+  @override
+  String rsaPssJwkAlg(HashImpl hash) {
+    if (hash == sha1) {
+      return 'PS1';
+    } else if (hash == sha256) {
+      return 'PS256';
+    } else if (hash == sha384) {
+      return 'PS384';
+    } else if (hash == sha512) {
+      return 'PS512';
+    } else {
+      throw UnsupportedError('hash is not supported');
+    }
+  }
+
+  @override
+  String rsassaPkcs1V15JwkAlg(HashImpl hash) {
+    if (hash == sha1) {
+      return 'RS1';
+    } else if (hash == sha256) {
+      return 'RS256';
+    } else if (hash == sha384) {
+      return 'RS384';
+    } else if (hash == sha512) {
+      return 'RS512';
+    } else {
+      throw UnsupportedError('hash is not supported');
+    }
+  }
 }
 
-class _Sha1 extends _Hash {
+final class _Sha1 extends _HashImpl {
   const _Sha1();
 
   @override
   ffi.Pointer<EVP_MD> Function() get _algorithm => ssl.EVP_sha1;
 }
```

Comment:
```suggestion
final class _Sha1 extends _HashImpl {
  const _Sha1();
  
  @override
  String get hmacJwkAlg => 'HS1';

  @override
  ffi.Pointer<EVP_MD> Function() get _algorithm => ssl.EVP_sha1;
}
```

This is actually a bit better too than using the `if-elseif`, if someone only uses `sha256` then the other classes can be omitted from the binary (by tree shaking). So the final binary will be a bit smaller! :tada: